sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SemanticModelHelper.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class SemanticModelHelper
+    {
+        public static SemanticModel GetCorrectSemanticModel(this SemanticModel model, SyntaxNode node)
```

Comment:
In addition to the overloads we would have to handle other methods like `GetTypeInfo` and also the fact that sometimes with `GetSymbolInfo` you want the symbol and sometimes you want the symbolInfo. So I prefer to keep returning the semantic model. About `GetSemanticModel` I think the name is misleading because `semanticModel.GetSemanticModel()` seems worse than `semanticModel.GetCorrectSemanticModel`. What about `GetSelfOrPartialNodeSemanticModel` or an extension method on `SyntaxNode` called `GetTiedSemanticModel`?